users/views.py

```diff
@@ -213,140 +218,193 @@ def verify_otp(request):
+@swagger_auto_schema(
+    operation_id='change_mail',
+    method= 'post',
+    request_body= ChangeMailSerializer,
+    responses={
+        '200': set_example(responses.change_mail_200),
+        '400': set_example(responses.forget_password_400),
+        '401': set_example(responses.verify_otp_401)
+    },
+)
+@api_view(['post'])
+@permission_classes([IsAuthenticated])
+def change_mail(request):
+    '''
+    To change the mail of the current authenticated user. OTP Verfication to be done in check_otp. 
+    '''
+    serializer = ChangeMailSerializer(data=request.data)
+    user = request.user
+
+    if serializer.is_valid():
+        new_email = serializer.validated_data['email']
+        # Changing the OTP and removing the verified status.
+        otp = send_email_otp([new_email])
+        user.otp = otp
+        user.email = new_email
+        user.verified = False
+        user.save()
+        # Verification to be done in check_otp.
+        return Response(responses.change_mail_200, status=status.HTTP_200_OK)
+
+    data = serializer.errors
+    return Response(data, status.HTTP_400_BAD_REQUEST)
     
-# TODO: remove ecell_user and client_check and use teesco auth
-@api_view(['GET'])
-@ecell_user
-@client_check
+
+
+@swagger_auto_schema(
+    operation_id='is_user_verified',
+    method= 'get',
+    responses={
+        '200': set_example({
+                    "verified": False,
+                }),
+        '401': set_example(responses.verify_otp_401)
+    },
+)
+@api_view(['get'])
+@permission_classes([IsAuthenticated])
 def is_user_verified(request):
-    user = request.ecelluser
+    '''
+    An API to check if the current user is verified or not.
+    '''
+    user = request.user
     verified = user.verified
     res_status = status.HTTP_200_OK
-    return Response({
-        "verified":verified,
-    }, status=res_status)
+
+    return Response(
+        {
+            "verified":verified,
+        }, 
+        status=res_status
+    )
 
 
 
-# TODO: remove ecell_user and client_check and use teesco auth
-@api_view(['GET'])
-@ecell_user
+@swagger_auto_schema(
+    operation_id='apply_for_ca',
+    method= 'get',
+    responses={
+        '200': set_example(responses.apply_for_ca_200),
+        '401': set_example(responses.verify_otp_401),
+
+    },
+)
+@api_view(['get'])
+@permission_classes([IsAuthenticated])
 def request_ca_approval(request):
-    res_status = status.HTTP_400_BAD_REQUEST
-    user = request.ecelluser
+    '''
+    API for current user to apply for Campus Ambassador program.
+    '''
+    user = request.user
+    if user.applied == True:
+        return Response({'message':'Already applied!'}, status.HTTP_200_OK)
     user.applied = True
     user.save()
-    message = "Congradulations. Applied for CA successfully"
     res_status = status.HTTP_200_OK
-    return Response({
-            "message": message,
-        }, status=res_status)
+    return Response(responses.apply_for_ca_200, status=res_status)
```

Comment:
```suggestion
    return Response({"message":"Congratulations! Applied for CA successfully."}, status=res_status)
```
Since `responses.apply_for_ca_200` has both the messages, sending that can be confusing to the user.